From ee4e7ce325556925f3ddff8faeab273c1513bbb2 Mon Sep 17 00:00:00 2001
From: "Fabian P. Schmidt" <kerel-fs@gmx.de>
Date: Sun, 19 Feb 2017 23:43:34 +0100
Subject: [PATCH 2/3] ogn-decode.service: Set StandardInput to tty

This is a workaround for #11: Version 0.2.5 crashs with SEGV if invoked by systemd,
since ogn-decode exits when it gets an EOF via stdin.
---
 scripts/ogn-decode.service | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/ogn-decode.service b/scripts/ogn-decode.service
index f2374e3..cc0448b 100644
--- a/scripts/ogn-decode.service
+++ b/scripts/ogn-decode.service
@@ -3,6 +3,12 @@ Description=Decode ogn/flarm data given by a fifo and forward it to glidernet.or
 After=network.target time-sync.target
 
 [Service]
+# Using a tty as StandardInput is needed as a workaround,
+# since ogn-decode requires an interactive tty (as in: 'doesn't return EOF'),
+# see also: https://github.com/glidernet/ogn-rf/issues/11
+StandardInput=tty-fail
+StandardOutput=journal
+TTYPath=/dev/tty3
 User=ogn
 ExecStart=/usr/bin/ogn-decode /etc/rtlsdr-ogn.conf
 Restart=on-failure
-- 
2.1.4

